atlasdb-api/src/test/java/com/palantir/atlasdb/keyvalue/api/CandidateCellForSweepingTest.java

```diff
@@ -0,0 +1,111 @@
+                    .collect(ImmutableList.toImmutableList());
+
+    private static final ImmutableList<Integer> SORTED_TIMESTAMPS_SIZES = ImmutableList.of(0, 1, 2, 100, 1000);
+
+    @Mock
+    private List<Long> MOCK_TIMESTAMPS;
+
+    @Test
+    public void candidateCellSizeWithLargerTimestampCollectionIsBigger() {
+        EXAMPLE_CELLS.forEach(cell -> {
+            CandidateCellForSweeping withOneTimestamp = createCandidateCell(cell, ImmutableSet.of(TIMESTAMP), true);
+            CandidateCellForSweeping withTwoTimestamps =
+                    createCandidateCell(cell, ImmutableSet.of(TIMESTAMP, TIMESTAMP + 1), false);
+            assertThat(withOneTimestamp.sizeInBytes()).isLessThan(withTwoTimestamps.sizeInBytes());
+        });
+    }
+
+    @Test
+    public void candidateCellSizeIsCorrectForDifferentSortedTimestampSizes() {
+        SORTED_TIMESTAMPS_SIZES.forEach(sortedTimestampsSize -> {
+            for (CandidateCellForSweeping candidate : createCandidateCells(sortedTimestampsSize)) {
+                assertThat(candidate.sizeInBytes())
+                        .isEqualTo(Long.sum(candidate.cell().sizeInBytes(), (long) sortedTimestampsSize * Long.BYTES));
+            }
+        });
+    }
+
+    @Test
+    public void noOverflowFromCollectionSize() {
+        // Mocking because otherwise we OOM.
+        when(MOCK_TIMESTAMPS.size()).thenReturn(Integer.MAX_VALUE);
+        Cell exampleCell = EXAMPLE_CELLS.get(0);
+        for (boolean isLatestValueEmpty : new boolean[] {true, false}) {
+            assertThat(createCandidateCell(exampleCell, MOCK_TIMESTAMPS, isLatestValueEmpty)
+                            .sizeInBytes())
+                    .isEqualTo(Long.sum(Integer.MAX_VALUE * 8L, exampleCell.sizeInBytes()));
+        }
+    }
+
+    private static ImmutableSet<CandidateCellForSweeping> createCandidateCells(int sortedTimestampsSize) {
+        ImmutableSet.Builder<CandidateCellForSweeping> builder = ImmutableSet.<CandidateCellForSweeping>builder();
+        for (boolean isLatestValueEmpty : new boolean[] {true, false}) {
+            builder.addAll(EXAMPLE_CELLS.stream()
+                    .map(cell -> createCandidateCell(
+                            cell, spawnCollectionOfTimestamps(sortedTimestampsSize), isLatestValueEmpty))
+                    .iterator());
+        }
+        return builder.build();
+    }
+
+    private static CandidateCellForSweeping createCandidateCell(
+            Cell cell, Collection<Long> sortedTimestamps, boolean isLatestValueEmpty) {
+        return ImmutableCandidateCellForSweeping.builder()
+                .cell(cell)
+                .sortedTimestamps(sortedTimestamps)
+                .isLatestValueEmpty(isLatestValueEmpty)
+                .build();
+    }
+
+    private static List<Long> spawnCollectionOfTimestamps(int size) {
+        return Collections.nCopies(size, TIMESTAMP);
+    }
+
+    private static byte[] spawnBytes(int size) {
```

Comment:
Out of curiosity, why do we need to fill the array with our default byte? I think all we really care about size, no?